samples/cellular/modem_shell/src/ping/icmp_ping.c

```diff
@@ -678,91 +712,127 @@ int icmp_ping_start(struct icmp_ping_shell_cmd_argv *ping_args)
+		inet_ntop(AF_INET6, &(ping_argv->current_addr6), src_ipv_addr,
 			  sizeof(src_ipv_addr));
 	}
 
 	st = getaddrinfo(src_ipv_addr, service, &hints, &res);
 	if (st != 0) {
-		ping_error(&ping_argv, "getaddrinfo(src) error: %d", st);
+		ping_error(ping_argv, "getaddrinfo(src) error: %d", st);
 		goto exit;
 	}
-	ping_argv.src = res;
+	ping_argv->src = res;
 
 	/* Get destination */
 	res = NULL;
 
-	st = getaddrinfo(ping_argv.target_name, service, &hints, &res);
+	st = getaddrinfo(ping_argv->target_name, service, &hints, &res);
 
 	if (st != 0) {
-		ping_error(&ping_argv, "getaddrinfo(dest) error: %d", st);
-		ping_error(&ping_argv, "Cannot resolve remote host\r\n");
-		freeaddrinfo(ping_argv.src);
+		ping_error(ping_argv, "getaddrinfo(dest) error: %d", st);
+		ping_error(ping_argv, "Cannot resolve remote host\r\n");
+		freeaddrinfo(ping_argv->src);
+		ping_argv->src = NULL;
 		goto exit;
 	}
-	ping_argv.dest = res;
-
-	if (ping_argv.src->ai_family != ping_argv.dest->ai_family) {
-		ping_error(&ping_argv, "Source/Destination address family error");
-		freeaddrinfo(ping_argv.dest);
-		freeaddrinfo(ping_argv.src);
+	ping_argv->dest = res;
+
+	if (ping_argv->src->ai_family != ping_argv->dest->ai_family) {
+		ping_error(ping_argv, "Source/Destination address family error");
+		freeaddrinfo(ping_argv->dest);
+		ping_argv->dest = NULL;
+		freeaddrinfo(ping_argv->src);
+		ping_argv->src = NULL;
 		goto exit;
 	}
 
 	struct sockaddr *sa;
 
-	sa = ping_argv.src->ai_addr;
-	ping_print(&ping_argv, "Source IP addr: %s", net_utils_sckt_addr_ntop(sa));
-	sa = ping_argv.dest->ai_addr;
-	ping_print(&ping_argv, "Destination IP addr: %s", net_utils_sckt_addr_ntop(sa));
+	sa = ping_argv->src->ai_addr;
+	ping_print(ping_argv, "Source IP addr: %s", net_utils_sckt_addr_ntop(sa));
+	sa = ping_argv->dest->ai_addr;
+	ping_print(ping_argv, "Destination IP addr: %s", net_utils_sckt_addr_ntop(sa));
 
 	/* Now we can check the max payload len for IPv6: */
-	uint32_t ipv6_max_payload_len = ping_argv.mtu - ICMP_IPV6_HDR_LEN - ICMP_HDR_LEN;
+	uint32_t ipv6_max_payload_len = ping_argv->mtu - ICMP_IPV6_HDR_LEN - ICMP_HDR_LEN;
 
-	if (ping_argv.src->ai_family == AF_INET6 &&
-	    ping_argv.len > ipv6_max_payload_len) {
+	if (ping_argv->src->ai_family == AF_INET6 &&
+	    ping_argv->len > ipv6_max_payload_len) {
 		ping_warn(
-			&ping_argv,
+			ping_argv,
 			"Payload size exceeds the link limits: MTU %d - headers %d = %d ",
-			ping_argv.mtu, (ICMP_IPV4_HDR_LEN - ICMP_HDR_LEN),
+			ping_argv->mtu, (ICMP_IPV4_HDR_LEN - ICMP_HDR_LEN),
 			ipv6_max_payload_len);
 		/* Continue still: */
 	}
-
-	icmp_ping_tasks_execute(&ping_argv);
-	return 0;
+	return true;
 exit:
 	if (pdp_context_info_tbl.array != NULL) {
 		free(pdp_context_info_tbl.array);
 	}
-	return -1;
+	return false;
 }
+
+/*****************************************************************************/
+
+int icmp_ping_start(struct icmp_ping_shell_cmd_argv *ping_args)
+{
+
+	icmp_ping_tasks_execute(ping_args);
```

Comment:
Could we just rename `icmp_ping_tasks_execute()` -> `icmp_ping_start()`?

Reply:
Maybe but for now I left it as it is.

Reply:
I'd imagine that this change would be small enough that we could do it now with very very little risks unless I'm missing something.

The reason I'm saying this is that I don't think it would be cleaned up any time soon and after two years there would be discussion whether this can be removed and why it's here...

Reply:
Yes, you got a point. Changes done. Please, recheck.